File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int c = 0;
        bool seenSpace = false;
        for(int i=s.length()-1;i>=0;i--){
            if(s[i] == ' ') seenSpace = true;
            if(s[i]!=' ') c++;
            if(seenSpace && s[i]==' ' && c!=0) break;
        }
        return c;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        seenSpace = False
        count = 0
        for i in range(len(s)-1,-1,-1):
            if s[i] == ' ': seenSpace = True
            if s[i] != ' ': count += 1
            if s[i] == ' ' and seenSpace and count != 0: break
        return count